Sources/ArcGISToolkit/Components/Offline/PreplannedMapModel.swift

```diff
@@ -223,6 +238,21 @@ extension PreplannedMapModel {
                 true
             }
         }
+        
+        /// A Boolean value indicating whether the local files can be removed.
+        var allowsRemoval: Bool {
+            switch self {
+            case .downloaded, .mmpkLoadFailure, .downloadFailure, .loadFailure, .packageFailure:
+                true
+            default:
+                false
+            }
+        }
+        
+        /// A Boolean value indicating whether the preplanned map area is downloaded.
+        var isDownloaded: Bool {
+            if case .downloaded = self { true } else { false }
+        }
```

Comment:
The convenience computed properties solely rely on the `Status` itself. If they are moved to the `Model` type, it would be more complicated to unit test the convenience properties against each individual status cases as now they rely on `model.status` property - which means I need to test each status in a different testcase.

Is that what you suggest here? I can make the change to both these properties and their tests.

Reply:
It is up to you. If it were me, I would prioritize convenience at the point of use by not putting these on the nested type. It may seem like it would make testing more complicated, but it doesn't need to. For example, you could model the model—or part of it—with a protocol and then put those convenience properties on the protocol, making them much easier to test. But like I said, it is up to you.